BridgeSDK/Networking/SBBNetworkManager.m

```diff
@@ -417,7 +417,15 @@ - (NSURLSessionDataTask *) doDataTask: (NSString*) method
   NSMutableURLRequest *request = [self requestWithMethod:method URLString:URLString headers:headers parameters:parameters error:nil];
     __block NSURLSessionDataTask *task = [self.mainSession dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
         NSError * httpError = [NSError generateSBBErrorForStatusCode:((NSHTTPURLResponse*)response).statusCode data:data];
-        NSDictionary * responseObject = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+        id responseObject = nil;
+        if (data.length) {
+            responseObject = [NSJSONSerialization JSONObjectWithData:data options:0 error:NULL];
+            if (!responseObject) {
+                // maybe it was an html error page
+                responseObject = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
```

Comment:
If it was an html error page, should it check for an httpError and create one if none exists? What if the server returns something else?

Current approach seems like there could be a problem without either of the error conditionals below getting called since this acts to handle an error like situation without actually creating an NSError object or validating that an NSError has been created that will later be handled.